text/0000-afp.md

```diff
@@ -0,0 +1,354 @@
+
+The default interaction between a requester and a farmer resembles the following:
+
+- Requester sends statement-of-work to farmer
+- Farmer validates sow and replies with quote for work
+- Requester validates quote and selects set of peers to complete work based on quotes
+- Requester sends agreement to selected farmers
+- Farmer validates agreement and countersigns agreement
+- Requester validates agreement and distributes work
+- When work is finished, requester sends reward to farmer
+- Farmer validates reward and returns a receipt to requester
+
+## 3. Background & Motivation
+
+In the current market, cryptocurrencies such as Bitcoin are oftentimes treated more as an investment than a currency. Although these digital assets are high in value, the lack of spending opportunities prevents them from being regularly exchanged for goods and services [[1]](#11-references). 
+
+In distributed services ecosystems, such as torrents, there is oftentimes the issue that there are more leechers (those who take from but do not contribute to a system) than there are seeders (those who support the system through being available to upload files) [[2]](#11-references). This is often attributed to the notion that there is no incentive to seed the network, since once you have the file, you likely do not gain from hosting it [[3]](#11-references).
+
+Ara Farming Protocol enables developers to provide incentives for users to participate in distributed services. Rewarding distributed services facilitates the circulation of tokens and gives a reason for peers to participate in the network. Instead of “mining”, which is generally associated with a task that does not inherently benefit the system, a user can earn tokens through “farming”, or participating in the service. AFP also helps instill trust in the work system by tying services to Ethereum smart contracts, in order to show full transparency when agreeing upon work.
+
+## 4. Ara Farming
+
+This section outlines terminology used throughout this document, as well expected methodologies needed for the protocol.
+
+### 4.1. Terminology
+
+#### 4.1.1. Farming
+
+Performing a service for another peer in the network, potentially for a reward.
+
+#### 4.1.2. Farmer
+
+A peer in the network who can provide a service.
+
+#### 4.1.3. Requester
+
+A peer in the network who desires to distribute work amongst peers.
+
+#### 4.1.4. AraId
+
+The Ara Identity associated with a peer.
+
+```proto
+message AraId {
+  string did = 1;
+}
+```
+
+#### 4.1.5. Signature
+
+A signed piece of data that can be used to validate a peer.
+
+```proto
+message Signature {
+  AraId ara_id = 1;
+  bytes data = 2;
+}
+```
+
+#### 4.1.6. Statement of Work (SOW)
+
+The scope of a distributable task, i.e. units-of-work, minimum resources, delivery date, etc.
+
+```proto
+message SOW {
+  bytes nonce = 1;
+  string work_unit = 2;
+  AraId requester = 3;
+  bytes data = 4;
+}
+```
+
+#### 4.1.7. Unit of Work
```

Comment:
Would be awesome to have a concrete example of this.  Maybe in the real world example?